xchange-okcoin/src/main/java/org/knowm/xchange/okcoin/service/OkCoinAccountService.java

```diff
@@ -44,8 +50,34 @@ public String withdrawFunds(Currency currency, BigDecimal amount, String address
 
   @Override
   public String requestDepositAddress(Currency currency, String... args) throws IOException {
-
     throw new NotAvailableFromExchangeException();
   }
 
+  @Override
+  public FundsInfo getFundsInfo(TradeHistoryParams params) throws ExchangeException, NotAvailableFromExchangeException, NotYetImplementedForExchangeException, IOException{
+    final OkCoinFundsInfoHistoryParams histParams = (OkCoinFundsInfoHistoryParams) params;
+    final OkCoinAccountRecords depositRecord = getAccountRecords(histParams.getSymbol(), "0",
+            String.valueOf(histParams.getPageNumber() != null ? histParams.getPageNumber() : 1),
+            String.valueOf(histParams.getPageLength() != null ? histParams.getPageLength() : 1));
+    final OkCoinAccountRecords withdrawalRecord = getAccountRecords(histParams.getSymbol(), "1",
+            String.valueOf(histParams.getPageNumber() != null ? histParams.getPageNumber() : 1),
+            String.valueOf(histParams.getPageLength() != null ? histParams.getPageLength() : 1));
+    final OkCoinAccountRecords[] okCoinAccountRecordsList = new OkCoinAccountRecords[] {depositRecord, withdrawalRecord};
+    return OkCoinAdapters.adaptFundsInfo(okCoinAccountRecordsList);
+  }
+
+  public static class OkCoinFundsInfoHistoryParams extends DefaultTradeHistoryParamPaging {
+
+    private final String symbol;
```

Comment:
I will email their OkCoin support now just to confirm the symbol format. Otherwise i would not know what parameter to use (currency pair or just currency). Logically it should just be Currency as you deposit/withdraw is based on Currency only. But right now, to get the BTC deposit/withdrawal history, i had to use Currency Pair (btc_usd). But to get USD history, i only need to use Currency (usd). I am not expert in trading so if I'm missing something out here please feel free to shed some light on me :). 

For now, what I will do just to avoid the hardcoding of string symbol, I will overload the adaptSymbol to accept Currency as well. And user should provide either of the two but not both.